iModelCore/ecobjects/src/ECInstance.cpp

```diff
@@ -4444,13 +4462,13 @@ bool IECInstance::SaveOnlyLoadedPropertiesToXml() const
 /*---------------------------------------------------------------------------------**//**
 * @bsimethod
 +---------------+---------------+---------------+---------------+---------------+------*/
-ECSchemaCP ECInstanceReadContext::FindSchemaCP(SchemaKeyCR key, SchemaMatchType matchType) const
+ECObjectsStatus ECInstanceReadContext::FindSchemaCP(SchemaKeyCR key, SchemaMatchType matchType, ECSchemaCP& schema) const
     {
-    ECSchemaCP schema = _FindSchemaCP(key, matchType);
-    if (NULL != schema)
-        return schema;
+    ECObjectsStatus status = _FindSchemaCP(key, matchType, schema);
+    if (ECObjectsStatus::Success != status)
+        schema = &m_fallBackSchema;

```

Comment:
It seems wrong to return the fallback schema when the schema we're looking for has been pruned

Reply:
That's what I pointed out in the very beginning...

Reply:
The updated code does not return fallback schema in case the schema we're looking for is pruned